Firmware/AVR/AVR_DRIVER/lib/encoder/encoder.c

```diff
@@ -0,0 +1,98 @@
+
+/**
+ * @file encoder.c
+ * @author Tsugumi Murata (tmurata293)
+ * @date 18 Feb 2021
+ * @brief library for encoder 
+ *
+ */
+
+#include <avr/io.h>
+#include <avr/interrupt.h>  
+#include "encoder.h"
+
+volatile int32_t encod_count           = 0x00;
+volatile encoder_state_t encod_curr    = ZERO_ZERO;
+volatile encoder_state_t encod_prev    = ZERO_ZERO;
+
+
+static encoder_state_t getCurrentPhase(){
```

Comment:
This is a good place to use inline. [STACKOVERFLOW: when to use inline](https://stackoverflow.com/questions/1932311/when-to-use-inline-function-and-when-not-to-use-it)
Note: it also depends on compiler optimization level O1-5. sometimes compiler would automatically convert function to inline whenever possible at >= O2 level, but its always a good practice to manually hint the compiler to do so. When optimization is disabled, the compiler ignore inline anyway.
```c
static inline encoder_state_E getCurrentPhase(){
```